### PR TITLE
[Feature] Add tidb datatype convertor

### PR DESCRIPTION
**Pull request by XiaoJiang521**
## Check list

* [ ] Code changed are covered with tests, or it does not need tests for reason:
* [ ] If any new Jar binary package adding in your PR, please add License Notice according
  [New License Guide](https://github.com/apache/seatunnel/blob/dev/docs/en/contribution/new-license.md)
* [ ] If necessary, please update the documentation to describe the new feature. https://github.com/apache/seatunnel/tree/dev/docs
* [ ] If you are contributing the connector code, please check that the following files are updated:
  1. Update change log that in connector document. For more details you can refer to [connector-v2](https://github.com/apache/seatunnel/tree/dev/docs/en/connector-v2)
  2. Update [plugin-mapping.properties](https://github.com/apache/seatunnel/blob/dev/plugin-mapping.properties) and add new connector information in it
  3. Update the pom file of [seatunnel-dist](https://github.com/apache/seatunnel/blob/dev/seatunnel-dist/pom.xml)
* [ ] Update the [`release-note`](https://github.com/apache/seatunnel/blob/dev/release-note.md).

### CODE REVIEW

**Review by ic4y — DISMISSED** [review]

**Comment by Hisoka-X** [comment]
Shall we add some test case for it?

**Comment by XiaoJiang521** [comment]
> Shall we add some test case for it?

If need to add test cases, I can also add, but I don't know how to test this，
and only Seatunnel Web uses this datatypeconvorter,

**Comment by Hisoka-X** [comment]
> > Shall we add some test case for it?
> 
> If need to add test cases, I can also add, but I don't know how to test this， and only Seatunnel Web uses this datatypeconvorter,

Got it.

File: seatunnel-connectors-v2/connector-jdbc/src/main/java/org/apache/seatunnel/connectors/seatunnel/jdbc/catalog/tidb/TiDBDataTypeConvertor.java [hunk 1]
```diff
@@ -0,0 +1,210 @@
...
+                dataTypeProperties = ImmutableMap.of(PRECISION, precision, SCALE, scale);
+                break;
+            default:
+                dataTypeProperties = Collections.emptyMap();
+                break;
+        }
+        return toSeaTunnelType(mysqlType, dataTypeProperties);
+    }
+
+    // todo: It's better to wrapper MysqlType to a pojo in ST, since MysqlType doesn't contains
```
Comment (Hisoka-X) [inline seatunnel-connectors-v2/connector-jdbc/src/main/java/org/apache/seatunnel/connectors/seatunnel/jdbc/catalog/tidb/TiDBDataTypeConvertor.java:85]: contains what?

**Review by Hisoka-X — APPROVED** [review]

File: seatunnel-connectors-v2/connector-jdbc/src/main/java/org/apache/seatunnel/connectors/seatunnel/jdbc/catalog/tidb/TiDBDataTypeConvertor.java (hunk 1 above)
Comment (XiaoJiang521) [reply seatunnel-connectors-v2/connector-jdbc/src/main/java/org/apache/seatunnel/connectors/seatunnel/jdbc/catalog/tidb/TiDBDataTypeConvertor.java:85]: contains properties. This place uses MysqlDataTypeConvertor, so take the annotations too QAQ

**Review by ruanwenjun — APPROVED** [review]
+1

**Pull request merged**